### PR TITLE
sql-mode(ticdc): dealing with SQL Mode more correctly

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?
Issue Number: close https://github.com/pingcap/tiflow/issues/10660

This PR is based on https://github.com/pingcap/tidb/pull/51321

### What is changed and how it works?
1. Delete the changefeed config `sql mode`
2. Clear the logic about dealing sql mode in dmls
3. Parse ddl with the sql mode in ddl job
4. Fix the session sql mode to the default sql mode when writing into downstream.
5. Add tests when setting sql mode in both upsteam and downstream

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/filter/expr_filter.go [hunk 1]
```diff
@@ -79,16 +77,10 @@ func newExprFilterRule(
 
 // verifyAndInitRule will verify and init the rule.
 // It should only be called in dmlExprFilter's verify method.
-func (r *dmlExprFilterRule) verify(tableInfos []*model.TableInfo, sqlMode string) error {
+func (r *dmlExprFilterRule) verify(tableInfos []*model.TableInfo) error {
```
Comment (CharlesCheung96) [inline pkg/filter/expr_filter.go:80]: Please add some comments to explain that the expr must be written in default sql mode.

**Review by asddongmen — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**